src/ray/common/task/task_spec.h

```diff
@@ -29,11 +29,52 @@
 extern "C" {
 #include "ray/thirdparty/sha256.h"
 }
-
 namespace ray {
-typedef ResourceSet SchedulingClassDescriptor;
 typedef int SchedulingClass;
 
+struct SchedulingClassDescriptor {
+ public:
+  explicit SchedulingClassDescriptor(ResourceSet rs, FunctionDescriptor fd, int64_t d)
+      : resource_set(std::move(rs)), function_descriptor(std::move(fd)), depth(d) {}
+  ResourceSet resource_set;
+  FunctionDescriptor function_descriptor;
+  int64_t depth;
+
+  bool operator==(const SchedulingClassDescriptor &other) const {
+    return depth == other.depth && resource_set == other.resource_set &&
+           function_descriptor == other.function_descriptor;
+  }
+
+  std::string DebugString() const {
+    std::stringstream buffer;
+    buffer << "{"
+           << "depth=" << depth << " "
+           << "function_descriptor=" << function_descriptor->ToString() << " "
+           << "resource_set="
+           << "{";
+    for (const auto &pair : resource_set.GetResourceMap()) {
+      buffer << pair.first << " : " << pair.second << ", ";
+    }
+    buffer << "}}";
```

Comment:
Do we show these in debug_state.txt (number of active tasks per scheduling class)?

Reply:
In this PR, we don't track the number of active tasks per scheduling class, but this debug string will be used once we do track it.